tests/vmi_lifecycle_test.go

```diff
@@ -571,6 +571,76 @@ var _ = Describe("[rfe_id:273][crit:high][vendor:cnv-qe@redhat.com][level:compon
 
 				tests.WaitForSuccessfulVMIStart(newVMI)
 			})
+
+			It("VMIs with Bridge Networking shouldn't fail after the kubelet restarts", func() {
+				bridgeVMI := vmi
+				// Remove the masquerade interface to use the default bridge one
+				bridgeVMI.Spec.Domain.Devices.Interfaces = nil
+				bridgeVMI.Spec.Networks = nil
+				v1.SetDefaults_NetworkInterface(bridgeVMI)
+				Expect(bridgeVMI.Spec.Domain.Devices.Interfaces).NotTo(BeEmpty())
+
+				By("starting a VMI with bridged network on a node")
+				bridgeVMI, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(bridgeVMI)
+				Expect(err).To(BeNil(), "Should submit VMI successfully")
+
+				// Start a VirtualMachineInstance with bridged networking
+				nodeName := tests.WaitForSuccessfulVMIStart(bridgeVMI)
+
+				tests.VerifyDummyNicForBridgeNetwork(bridgeVMI)
+
+				By("restarting kubelet")
+				pod := renderPkillAllPod("kubelet")
+				pod.Spec.NodeName = nodeName
+				_, err = virtClient.CoreV1().Pods(tests.NamespaceTestDefault).Create(pod)
+				Expect(err).ToNot(HaveOccurred())
+
+				By("starting another VMI on the same node, to verify kubelet is running again")
+				newVMI := newCirrosVMI()
+				newVMI.Spec.NodeSelector = map[string]string{"kubernetes.io/hostname": nodeName}
+				Eventually(func() error {
+					newVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(newVMI)
+					Expect(err).To(BeNil())
+					return nil
+				}, 100, 10).Should(Succeed(), "Should be able to start a new VM")
+
+				By("checking if the VMI with bridged networking is still running, it will verify the CNI didn't cause the pod to be killed")
```

Comment:
Let me pull out my notes....

When I dug through the Kubelet code previously, I saw that the Kubelet does an [RPC call](https://github.com/containernetworking/cni/blob/master/SPEC.md#parameters) `CHECK` to the CNI.  If `CHECK` is implemented, it goes through the pods to see if a pod's interface & IP == a previous result.  Most CNIs don't implement `CHECK` but [multus ](https://github.com/intel/multus-cni/blob/d4e86998253d2c0d44f3d8317887ff7c926e7756/pkg/multus/multus.go#L880) does.  So Kubelet isn't going to kill the pod unless you have a CNI with `CHECK` implemented.  Maybe the language should be more specific on this point?  Something like:
```
VMs using bridged networking will survive a kubelet restart by having kubevirt create a dummy interface on the virt-launcher pods, so that some Kubernetes CNIs, that have implemented the `CHECK` RPC call, will not cause VMI pods to enter a failed state.
```

```
checking if the VMI with bridged networking is still running.   Some CNIs will cause pods to be in a failed state
```

Reply:
I guess it depends on how you see it. The last time I checked the code, we simply don't get an IP back (because there is none configured). The kubelet then says: hey something is wrong, this pod should be running and happily have the network properly configured already. Something is wrong, let's tear it down.

I don't mind what we write. It may change anyway depending on the k8s version. :)